### PR TITLE
add forceMediaItemWidthBasedOnMinClips config

### PR DESCRIPTION
**Pull request by vladm3**
add forceMediaItemWidthBasedOnMinClips config to make playlistAPI not to override mediaItemWidth if items number is less than MinClips config

### CODE REVIEW

**Comment by eitanavgil** [comment]
@itaykinnrot 
https://kaltura.atlassian.net/browse/FEC-4314

File: modules/KalturaSupport/components/playlistAPI.js [hunk 1]
```diff
@@ -325,7 +326,13 @@
 					}
 				}else{
 					var width = this.getConfig("fixedControls") ? $( ".playlistInterface" ).width() - this.getConfig("horizontalControlsWidth")*2 : $( ".playlistInterface" ).width();
-					this.setConfig( 'mediaItemWidth', Math.floor(width / this.getConfig("MinClips")) );
+					// decide width of the items
+					// If should fill remaining horizontal space use preconfigured 'MinClips'
+					// else use cached initial 'MinClips' config
+					var itemsNumber = this.getConfig('horizontalFillRemainingSpace') ?
+						this.getConfig('MinClips') : this.minClips;
```
Comment (amirch1) [inline modules/KalturaSupport/components/playlistAPI.js]: @vladm3 @eitanavgil  - I don't understand the logic here. as far as I can tell this.getConfig('MinClips') and this.minClips return the same value.
@eitanavgil  - lets discuss over Skype

Comment (eitanavgil) [inline modules/KalturaSupport/components/playlistAPI.js]: @vladm3 can you please respond?

Comment (vladm3) [inline modules/KalturaSupport/components/playlistAPI.js]: @amirch1 - this.getConfig('MinClips') stores current minClips config which may not be equal to the initial minClips passed to playlist plugin, on the other hand, this.minClips stores initial minClips value passed to the plugin (it's assigned in setup method here: https://github.com/vladm3/mwEmbed/blob/2.37-width-pl-param/modules/KalturaSupport/components/playlistAPI.js#L69)

**Comment by eitanavgil** [comment]
@amirch1 does that makes sense? it does fixes our case where the list is dynamic and is being changed on the fly when we search in one of our plugins and injects new playlist content.

**Comment by eitanavgil** [comment]
@amirch1 ping?

**Comment by amirch1** [comment]
@eitanavgil , @vladm3  - sorry, it still doesn't make sense to me. I guess it applies only to dynamic playlists. When I'm creating a regular horizontal playlist with MinClips bigger the the actual number of items and horizontalFillRemainingSpace: true, the items still don't fill the playlist width. I'd rather have a general fix for this issue.I've created a pull request on 2.38:
https://github.com/kaltura/mwEmbed/pull/2345/files
This works with regular playlists. Please check if this solves the dynamic playlist issue as well and let me know if I can merge it or if you want to add something to it. Thanks.

**Pull request closed without merging**